pylint/checkers/refactoring.py

```diff
@@ -261,13 +266,48 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        if self._is_iterating_mutable_sequence(node):
+            parts = node.nodes_of_class(astroid.Call)
+            if isinstance(parts, collections.Iterable):
+                for part in parts:
+                    if self._check_part_is_call_method_elimination(node.iter,
```

Comment:
What is `part` in this function name.

Maybe something along a lines of `method_call_mutates_sequence(sequence, method)` is better signature? (but I'm not native speaker, so someone should confirm...)